### ISSUE TITLE
Regression: --enable config is being ignored

### ISSUE THREAD

**Issue by itamarst** [PyCQA/pylint#4574]
### Steps to reproduce

Let's say you have `example.py`:

```python
l = []
for x in range(10):
    l.append(lambda: x + 10)
```

I only want to run `cell-var-from-loop` check.

If I run `pylint` with no config, it's fine:

```
$ pylint example.py 
************* Module example
example.py:1:0: C0114: Missing module docstring (missing-module-docstring)
example.py:3:21: W0640: Cell variable x defined in loop (cell-var-from-loop)

-------------------------------------------------------------------
Your code has been rated at 3.33/10 (previous run: 10.00/10, -6.67)
```

### Current behavior

If I run pylint in a way that is supposed to only check `cell-var-from-loop` it reports... no problems!

```
$ pylint --disable=all --enable=cell-var-from-loop example.py 

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 3.33/10, +6.67)
```

The `--enable` appears to be ignored.

### Expected behavior

I would expect the `cell-var-from-loop` check to be run. The same thing happens with `W0640`.

### pylint --version output

Result of `pylint --version` output:

```
$ pylint --version
pylint 2.8.3
astroid 2.5.8
Python 3.6.13 (default, Feb 16 2021, 00:00:00) 
[GCC 10.2.1 20201125 (Red Hat 10.2.1-9)]
```

**Comment by itamarst** [comment]
Pylint 2.7, 2.6, 2.5 have the same issue. Pylint 2.4 does the right thing:

```
$ pylint --version
pylint 2.4.4
astroid 2.3.3
Python 3.6.13 (default, Feb 16 2021, 00:00:00) 
[GCC 10.2.1 20201125 (Red Hat 10.2.1-9)]
$ pylint --disable=all --enable=cell-var-from-loop example.py 
************* Module example
example.py:3:21: W0640: Cell variable x defined in loop (cell-var-from-loop)

-------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 10.00/10, -3.33)
```

**Comment by Pierre-Sassoulas** [comment]
Hello, thank you for reporting this issue and starting to analyse the problem.

**Comment by Pierre-Sassoulas** [comment]
I did a git bisect, the regression appeared in c5181886b59799740306829e0e21e548c50db391

**Comment by jacobtylerwalls** [comment]
Fixed in 2.10 (678d4f10ae3fcd9396360a5cff3e084aae31e830)

**Comment by Pierre-Sassoulas** [comment]
Reopening because I had a regression test ready :)

**Comment by itamarst** [comment]
Thank you!
